src/orange/surf/Involute.cc

```diff
@@ -10,17 +10,35 @@
 namespace celeritas
 {
 //---------------------------------------------------------------------------//
+/*
+ * \parma radius: radius of the circle of involute
+ * \param a: displacement angle of the involute
+ * \param sign: chirality of involute
+ * \param tmin: minimum tangent angle
+ * \param tmax: maximum tangent angle
+ */
+
 Involute::Involute(Real3 const& origin,
                    real_type radius,
                    real_type a,
-                   real_type sign,
                    real_type tmin,
                    real_type tmax)
-    : origin_(origin), r_b_(radius), a_(a), sign_(sign), tmin_(tmin), tmax_(tmax)
+    : origin_(origin)
+    , r_b_(std::fabs(radius))
+    , a_(a)
+    , sign_(static_cast<Sign>(radius < 0))
+    , tmin_(tmin)
+    , tmax_(tmax)
 {
-    CELER_EXPECT(radius > 0);
     CELER_EXPECT(a > 0);
-    CELER_EXPECT(std::fabs(tmax) < 2 * pi + std::fabs(tmin));
+    CELER_EXPECT(tmax > 0);
+    CELER_EXPECT(tmin > 0);
+    CELER_EXPECT(tmax < 2 * pi + tmin);
+
+    if (sign_)
+    {
+        a_ = -a + pi;
```

Comment:
Or save an operation...
```suggestion
        a_ = pi - a;
```